src/stream.rs

```diff
@@ -78,36 +70,37 @@ where
 
 impl<T> Stream for EventStream<T>
 where
-    T: AsMut<[u8]> + AsRef<[u8]>,
+    T: AsMut<[u8]> + AsRef<[u8]> + Unpin,
 {
-    type Item = EventOwned;
-    type Error = io::Error;
+    type Item = io::Result<EventOwned>;
 
-    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error>
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>>
     {
-        if self.unused_bytes == 0 {
+        let self_ = self.get_mut();
```

Comment:
Can we use [`get_unchecked_mut`](https://doc.rust-lang.org/std/pin/struct.Pin.html#method.get_unchecked_mut) here, to forgo the requirement for `Unpin`? I don't think we're moving out of `self_`, so it should be fine, I think.